File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/ChainedMessageListener.java
package com.github.bumblebee.command;

import telegram.domain.Update;
import telegram.polling.UpdateHandler;

public abstract class ChainedMessageListener implements UpdateHandler {

    public abstract boolean onMessage(Long chatId, String message, Update update);

    @Override
    public boolean onUpdate(Update update) {
        Long chatId = update.getMessage().getChat().getId();
        String text = update.getMessage().getText();

        return onMessage(chatId, text, update);
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/ChainedMessageListener.kt
package com.github.bumblebee.command

import com.github.bumblebee.bot.consumer.UpdateHandler
import com.github.telegram.domain.Update

abstract class ChainedMessageListener : UpdateHandler {

    abstract fun onMessage(chatId: Long, message: String?, update: Update): Boolean

    override fun onUpdate(update: Update): Boolean {
        // 'edits' and other types of updates aren't supported for now
        val message = update.message
        if (message != null) {
            val chatId = message.chat.id
            val text = message.text

            return onMessage(chatId, text, update)
        }
        return false
    }
}
